tools/configuration-converter.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to convert istio configuration to sail operator configuration.
+# In the end of the execution new yaml file will be created with "sail-ISTIO_CONFIG_YAML" name.
+# Usage: ./configuration-converter.sh ISTIO_CONFIG_YAML_WITH_PATH, example:  ./configuration-converter.sh sample_config.yaml"
+set -e
+
+# Function to show usage
+usage() {
+    echo "Usage: $0 <input> [output] [-n <namespace>] [-v <version>]"
+    echo "  <input>         : Input file (required)"
+    echo "  [output]        : Output file (optional, defaults to input's base directory with '-sail.yaml' suffix)"
+    echo "  -n <namespace>  : Namespace (optional, defaults to 'istio-system')"
+    echo "  -v <version>    : Istio Version (optional)"
+    exit 1
+}
+
+# Ensure at least one file argument is provided
+if [[ $# -lt 1 ]]; then
+    usage
+fi
+
+# Initialize variables
+INPUT=""
+OUTPUT=""
+NAMESPACE="istio-system"
+VERSION=""
+
+# Identify the first and second file paths (treat the first file as INPUT)
+for arg in "$@"; do
+    if [[ -f "$arg" && -z "$INPUT" ]]; then
+        INPUT="$arg"  # First file is the INPUT
+    elif [[ -f "$arg" && -z "$OUTPUT" ]]; then
+        OUTPUT="$arg"  # Second file is the OUTPUT
+    fi
+done
+
+# Ensure the input file is provided and is valid
+if [[ -z "$INPUT" || ! -f "$INPUT" ]]; then
+    echo "Error: Input file is missing or invalid."
+    usage
+fi
+
+# If OUTPUT is not specified, generate a default output file name
+if [[ -z "$OUTPUT" ]]; then
+    OUTPUT="$(dirname "$INPUT")/$(basename "$INPUT" .yaml)-sail.yaml"
+elif [[ -d "$OUTPUT" ]]; then
+    echo "Error: OUTPUT must be a file, not a directory."
+    exit 1
+fi
+
+# Parse optional namespace (-n) and version (-v) arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -n)
+            NAMESPACE="$2"
```

Comment:
Nit:  
If the user accidentally supplies "-n" without the value (similarly for version), it will not work as we are not checking the value of $2.